tron/core/actionrun.py

```diff
@@ -653,6 +653,47 @@ def submit_command(self):
         mesos_cluster.submit(task)
         return task
 
+    def recover(self):
+        if self.mesos_task_id is None:
+            log.error(f'No task ID, cannot recover {self}')
+            return
+
+        okay_to_recover = self.machine.transition('running')
```

Comment:
I don't really understand this. Isn't it premature to transition the machine to `running` before we have even talked to mesos?

Shouldn't we talk to mesos first and make sure it is in `TASK_RUNNING` before moving the state machine?

Reply:
But we are processing all these runs serially anyway, the actual transition happens in this same function.

How about running `check('running')` instead?

The thing I'm worried about is all the things that could go wrong later in the function, and I think that moving it right to "running" as the first thing we do in this function is presumptuous. We haven't even talked to mesos yet?

Reply:
For monitoring concerns, what if the script just checked the uptime of tron first? If it is <2m say, just skip out and give it time to settle in before checking jobs. (will be nice to go easy on the api anyway to give it some breathing room after a restart)

Reply:
I moved the transition to the end of the method, so it doesn't transition if something goes wrong before submitting to taskproc.

At that point, I think it's fair to be optimistic and say it's running - we do the same for our SSH recovery. It helps with monitoring, and also matters if the job has `allow_overlap: False` (unknown wouldn't prevent another run from starting).